packages/amplify_api/lib/method_channel_api.dart

```diff
@@ -160,6 +160,13 @@ class AmplifyAPIMethodChannel extends AmplifyAPI {
             AmplifyExceptionMessages.nullReturnedFromMethodChannel);
       final errors = _deserializeGraphQLResponseErrors(result);
 
+      // Catch data while its non-nullable and provide error messages
+      if (result['data'] == null) {
```

Comment:
We may need to brainstorm another way to handle this case. There may not always be an error in `errors` if this is `null` - sometimes null is a valid response.

Reply:
This goes back to changing the GraphQLRequest to accept nullable data. Currently null responses will break workflows and I've found the error reporting for these errors is buried too deep. For now I think this will at least provide developers (mainly internal) with missing error context until a more permanent solution is reached.

Reply:
Oh that's right - sorry I have the memory of a goldfish!